pkg/network/probe_handler_test.go

```diff
@@ -96,8 +96,46 @@ func TestProbeHandlerSuccessfulProbe(t *testing.T) {
 				t.Errorf("prober.Do() = nil, expected an error")
 			}
 			if got != c.want {
-				t.Errorf("unexpected probe result: want: %t, got: %t", c.want, got)
+				t.Errorf("Unexpected probe result: want: %t, got: %t", c.want, got)
 			}
 		})
 	}
 }
+
+func BenchmarkProbeHandler(b *testing.B) {
+	var h http.Handler = http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.Write([]byte("Inner Body"))
+	})
+	h = NewProbeHandler(h)
+	ts := httptest.NewServer(h)
+	defer ts.Close()
+	options := []interface{}{
+		prober.ExpectsStatusCodes([]int{http.StatusOK}),
+	}
+
+	b.Run(fmt.Sprint("sequential"), func(b *testing.B) {
+		for j := 0; j < b.N; j++ {
+			got, err := prober.Do(context.Background(), network.AutoTransport, ts.URL, options...)
```

Comment:
You're not testing the handler as the test's location suggests. This is testing the prober itself, which is not what we want to do.

Instead, we should run the handler itself using a manually constructed, static HTTP request (as it would be created by `prober.Do` but without actually using it). The results should be close to or equal to 0 allocations in this case.